Source/Classes/SLKTypingIndicatorProtocol.h

```diff
@@ -0,0 +1,46 @@
+//
+//   Copyright 2014 Slack Technologies, Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+/**
+ Generic protocol needed when customizing your own typing indicator view.
+ SLKTextViewController depends of the 'isVisible' property internally, to detect with KVO its changes and update the typing indicator view's constraints automatically.
+ Because of this, you MUST conform to this protocol and implement the required methods.
+ */
+@protocol SLKTypingIndicatorProtocol <NSObject>
+@required
+
+/** Returns YES if the indicator is visible. */
+@property (nonatomic, getter = isVisible) BOOL visible;
+
+/**
+ Updates the 'visible' state.
+ To enable the typing indicator, you MUST override this method and implement key-value observer compliance manually,
+ using the -willChangeValueForKey: and -didChangeValueForKey: methods.
+ 
+ @return YES if the indicator view is visible.
+ */
+- (void)setVisible:(BOOL)visible;
```

Comment:
LOL. No, what are the method and/or var signatures that shut up the compiler about not complying to the protocol? Swift doesn't support different names for getters/setters. I don't know how to create write-only variable. Creating functions doesn't seem to work. I have no idea how Swifts "sees" the protocol definition and how that translates into Swift. I'd like to have a "fix-me" on those non-conforming diagnostics, and have the UI automatically fill in placeholder implementations for me, instead of guessing how Swift wants the methods to look like.